### PR TITLE
Wizard spell logging now uses the actual name of the spell

### PR DESCRIPTION
**Pull request by ChangelingRain**
Adds logging for wizard spell upgrades
Closes #25985

### CODE REVIEW

**Comment by LanCartwright** [comment]
Thank you for this Joan; I hope this will make tracking spell usage easier in the future and less confusing.

**Comment by XDTM** [comment]
Why was it like that in the first place

**Comment by XDTM** [comment]
And why would it close #25985

**Comment by ChangelingRain** [comment]
it conflicts and it's easier to steal the changes and do it myself.

**Comment by XDTM** [comment]
Oh i missed the change in the diff

File: code/game/gamemodes/wizard/spellbook.dm [hunk 1]
```diff
@@ -58,9 +57,10 @@
 						aspell.name = "Instant [aspell.name]"
 				if(aspell.spell_level >= aspell.level_max)
 					to_chat(user, "<span class='notice'>This spell cannot be strengthened any further.</span>")
+				feedback_add_details("wizard_spell_improved", "[name]|[aspell.level]")
 				return 1
 	//No same spell found - just learn it
-	feedback_add_details("wizard_spell_learned",log_name)
+	feedback_add_details("wizard_spell_learned", name)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/wizard/spellbook.dm]: "[name]"

File: code/game/gamemodes/wizard/spellbook.dm [hunk 2]
```diff
@@ -294,7 +265,7 @@
 
 /datum/spellbook_entry/item/Buy(mob/living/carbon/human/user,obj/item/weapon/spellbook/book)
 	new item_path(get_turf(user))
-	feedback_add_details("wizard_spell_learned",log_name)
+	feedback_add_details("wizard_spell_learned", name)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/wizard/spellbook.dm]: ^

File: code/game/gamemodes/wizard/spellbook.dm [hunk 3]
```diff
@@ -513,7 +465,7 @@
 		return TRUE
 
 /datum/spellbook_entry/summon/ghosts/Buy(mob/living/carbon/human/user, obj/item/weapon/spellbook/book)
-	feedback_add_details("wizard_spell_learned", log_name)
+	feedback_add_details("wizard_spell_learned", name)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/wizard/spellbook.dm]: ^

File: code/game/gamemodes/wizard/spellbook.dm [hunk 4]
```diff
@@ -523,15 +475,14 @@
...
-	log_name = "SG"
 
 /datum/spellbook_entry/summon/guns/IsAvailible()
 	if(!SSticker.mode) // In case spellbook is placed on map
 		return 0
 	return (SSticker.mode.name != "ragin' mages" && !config.no_summon_guns)
 
 /datum/spellbook_entry/summon/guns/Buy(mob/living/carbon/human/user,obj/item/weapon/spellbook/book)
-	feedback_add_details("wizard_spell_learned",log_name)
+	feedback_add_details("wizard_spell_learned", name)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/wizard/spellbook.dm]: ^

File: code/game/gamemodes/wizard/spellbook.dm [hunk 5]
```diff
@@ -541,15 +492,14 @@
...
-	log_name = "SU"
 
 /datum/spellbook_entry/summon/magic/IsAvailible()
 	if(!SSticker.mode) // In case spellbook is placed on map
 		return 0
 	return (SSticker.mode.name != "ragin' mages" && !config.no_summon_magic)
 
 /datum/spellbook_entry/summon/magic/Buy(mob/living/carbon/human/user,obj/item/weapon/spellbook/book)
-	feedback_add_details("wizard_spell_learned",log_name)
+	feedback_add_details("wizard_spell_learned", name)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/wizard/spellbook.dm]: ^

File: code/game/gamemodes/wizard/spellbook.dm [hunk 6]
```diff
@@ -569,7 +518,7 @@
 	return (SSticker.mode.name != "ragin' mages" && !config.no_summon_events)
 
 /datum/spellbook_entry/summon/events/Buy(mob/living/carbon/human/user,obj/item/weapon/spellbook/book)
-	feedback_add_details("wizard_spell_learned",log_name)
+	feedback_add_details("wizard_spell_learned", name)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/wizard/spellbook.dm]: ^

**Comment by ExcessiveUseOfCobblestone** [comment]
WAIT IM RETARDED HAHAHA

**Review by ExcessiveUseOfCobblestone — APPROVED** [review]

**Comment by Cheridan** [comment]
oops shit

File: code/game/gamemodes/wizard/spellbook.dm [hunk 7]
```diff
@@ -504,7 +456,7 @@
 /datum/spellbook_entry/summon/ghosts
 	name = "Summon Ghosts"
 	desc = "Spook the crew out by making them see dead people. Be warned, ghosts are capricious and occasionally vindicative, and some will use their incredibly minor abilties to frustrate you."
-	log_name = "SGH"
+	cost = 0
```
Comment (Shadowlight213) [inline code/game/gamemodes/wizard/spellbook.dm]: Isn't this a balance change more than a fix?

Comment (ChangelingRain) [inline code/game/gamemodes/wizard/spellbook.dm]: See the linked pr in the OP.

Comment (Shadowlight213) [inline code/game/gamemodes/wizard/spellbook.dm]: No idea why coiax thought it cost 0 points originally. I looked at the original pr adding it, and it clearly cost 2 points at the time.

File: code/game/gamemodes/wizard/spellbook.dm [hunk 8]
```diff
@@ -484,6 +435,7 @@
 	category = "Rituals"
 	refundable = 0
 	buy_word = "Cast"
+	cost = 2
```
Comment (Shadowlight213) [inline code/game/gamemodes/wizard/spellbook.dm]: This is pointless. The parent spellbook_entry has the cost set to 2. There is no reason to set it again.

**Review by Shadowlight213 — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/wizard/spellbook.dm (hunk 8 above)
Comment (ChangelingRain) [inline code/game/gamemodes/wizard/spellbook.dm]: I just copied coiax's changes~

Comment (Shadowlight213) [inline code/game/gamemodes/wizard/spellbook.dm]: \>blindly copying others changes.
smh

Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/wizard/spellbook.dm]: Coiax of all people!

**Pull request merged**